### PR TITLE
Add Feature Design Consideration section to DG

### PR DESCRIPTION
**Pull request by UncleGrandpa925**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -60,7 +60,7 @@ By combining these 4 core functions into a single app, we are able to deliver a
 * View all contacts
 * View all module-specific contacts (i.e. Professors, TAs, friends in the same module)
 
-Any help on the development of TrackIt@NUS would be greatly appreciated, and there are several ways to do so:
+Any helps on the development of TrackIt@NUS would be greatly appreciated, and there are several ways to do so:
```
Comment (simonteozw) [inline docs/DeveloperGuide.md]: Shouldn't be helps, should just be help

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -301,15 +301,113 @@ A possible drawback of this uniform design is that it may not be the most approp
...
+
+Similar to the previous module edit, one clear drawback of implementing this behavior is it will complicate the app's logic and make the app more bug-prone.
+
+**After having considered all the pros & cons, we have decided to implement this feature so that the user can have the best possible User Experience!**
+
+----
+
+**Allow users to use mouse to click on tabs**
+
+_This feature is one of the most debated feature in our team._
```
Comment (simonteozw) [inline docs/DeveloperGuide.md]: one of the most debated features, needs the s

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -301,15 +301,113 @@ A possible drawback of this uniform design is that it may not be the most approp
...
+----
+
+**Allow users to use mouse to click on tabs**
+
+_This feature is one of the most debated feature in our team._ 
+
+We are well aware of the module's constraints that the app must be optimized for CLI. 
+
+Yet, 5 todo-apps have been studied by our team (macOS's reminders, Microsoft To Do, Todoist, OmniFocus) and we found all of these apps only have one way to change tabs, and it is by clicking on them. 
+Although forcing users to use CLI to change tab can be easily implemented, it will make the app unintuitive to most users (except CLI guru), and hence **cannot attract new users** to change to our new app from their old apps(of which they are so used to clicking on tabs).
```
Comment (simonteozw) [inline docs/DeveloperGuide.md]: CLI gurus not guru, needs the s

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -301,15 +301,113 @@ A possible drawback of this uniform design is that it may not be the most approp
...
+| add task | `T add n/NAME d/DATE [m/CODE] [r/REMARK]` | 
+| add contact | `C add n/NAME p/PHONE_NUMBER e/EMAIL [t/TAG]...` | 
+
+| Command | Format |
+| -- | -------- |
+| edit module | `M edit CODE [m/NEW_CODE] [n/NAME]` | 
+| edit lesson | `L edit INDEX [m/CODE] [t/TYPE] [d/DATE] [a/ADDRESS]` | 
+| edit task | `T edit INDEX [n/NAME] [d/DATE] [m/CODE] [r/REMARK]` | 
+| edit contact | `C edit INDEX [n/NAME] [p/PHONE_NUMBER] [e/EMAIL] [t/TAG]...` |
+
```
Comment (simonteozw) [inline docs/DeveloperGuide.md:378]: Hmm, should this be under code design considerations? Not sure if command syntax is a feature?

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -301,15 +301,113 @@ A possible drawback of this uniform design is that it may not be the most approp
...
+| Command | Format |
+| -- | -------- |
+| help | `help` |
+| exit | `exit` |
+
+This will relieve our users from the mental burden of having to remember different syntax for different commands. Although there are only 14 commands to get used to, having to remember all 14 of them will be a huge barrier for our new potential users and also, it will put an unnecessary mental burden on the users of our app.
+
+----
+**Other small feature consideration**
```
Comment (simonteozw) [inline docs/DeveloperGuide.md]: considerations

**Review by simonteozw — COMMENTED** [review]
Looks good! Just a few comments

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (UncleGrandpa925) [reply docs/DeveloperGuide.md:378]: it is a feature right? It doesn't affect the back-end

File: docs/DeveloperGuide.md (hunk 5 above)
Comment (UncleGrandpa925) [reply docs/DeveloperGuide.md]: Okay will fix it all

Comment (UncleGrandpa925) [reply docs/DeveloperGuide.md]: Done! Can you review it again?

**Review by simonteozw — APPROVED** [review]
LGTM

**Pull request merged**